tools/apiview/parsers/csharp-api-parser/Program.cs

```diff
@@ -0,0 +1,145 @@
+using System.CommandLine;
+using System.IO.Compression;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Xml.Linq;
+using ApiView;
+
+var inputOption = new Option<FileInfo>("--packageFilePath", "C# Package (.nupkg) file").ExistingOnly();
+inputOption.IsRequired = true;
+
+var outputOption1 = new Option<DirectoryInfo>("--outputDirectoryPath", "Directory for the output Token File").ExistingOnly();
+var outputOption2 = new Option<string>("--outputFileName", "Output File Name");
+var runAnalysis = new Argument<bool>("runAnalysis", "Run Analysis on the package");
+runAnalysis.SetDefaultValue(true);
+
+var rootCommand = new RootCommand("Parse C# Package (.nupkg) to APIView Tokens")
+{
+    inputOption,
+    outputOption1,
+    outputOption2,
+    runAnalysis
+};
+
+rootCommand.SetHandler((FileInfo packageFilePath, DirectoryInfo outputDirectory, string outputFileName, bool runAnalysis) =>
+{
+    try
+    {
+        using (var stream = packageFilePath.OpenRead())
+        {
+            HandlePackageFileParsing(stream, packageFilePath, outputDirectory, outputFileName, runAnalysis);
+        }
+    }
+    catch (Exception ex)
+    {
+        Console.Error.WriteLine($"Error Reading PackageFile : {ex.Message}");
+    }
+}, inputOption, outputOption1,outputOption2, runAnalysis);
+
+return rootCommand.InvokeAsync(args).Result;
+
+
+static void HandlePackageFileParsing(Stream stream, FileInfo packageFilePath, DirectoryInfo OutputDirectory, string outputFileName, bool runAnalysis)
```

Comment:
This can potentially be simplified with the `NuGet.Protocol` package - samples [here](https://github.com/NuGet/Samples/blob/main/NuGetProtocolSamples/Program.cs#L200)